cnd/src/network.rs

```diff
@@ -310,49 +314,76 @@ impl ComitNode {
         Ok(())
     }
 
-    pub fn take_hbit_herc20_order(
+    /// The taker plays the role of Alice.
+    pub fn take_herc20_hbit_order(
         &mut self,
         order_id: OrderId,
         swap_id: LocalSwapId,
-        refund_address: crate::bitcoin::Address,
-        refund_identity: identity::Bitcoin,
-        redeem_identity: identity::Ethereum,
+        redeem_identity: crate::bitcoin::Address,
+        refund_identity: identity::Ethereum,
     ) -> anyhow::Result<()> {
-        let local_data = LocalData {
+        let transient = self
+            .storage
+            .derive_transient_identity(swap_id, Role::Alice, Side::Beta);
+
+        self.bitcoin_addresses.insert(transient, redeem_identity);
+
+        let data = LocalData {
             secret_hash: Some(SecretHash::new(
                 self.seed.derive_swap_seed(swap_id).derive_secret(),
             )),
-            ethereum_identity: Some(redeem_identity),
+            ethereum_identity: Some(refund_identity),
+            bitcoin_identity: Some(transient),
             lightning_identity: None,
-            bitcoin_identity: Some(refund_identity),
         };
+        self.local_data.insert(swap_id, data);
 
-        self.local_data.insert(swap_id, local_data);
+        self.order_swap_ids.insert(order_id, swap_id);
+        self.orderbook.take(order_id)?;
 
-        self.comit
-            .take_order(order_id, refund_address, redeem_identity)
+        Ok(())
     }
 
-    pub fn make_hbit_herc20_order(
+    /// The maker plays the role of Bob.
+    pub fn make_herc20_hbit_order(
         &mut self,
         order: NewOrder,
-        refund_identity: crate::bitcoin::Address,
+        swap_id: LocalSwapId,
         redeem_identity: identity::Ethereum,
+        refund_identity: crate::bitcoin::Address,
     ) -> anyhow::Result<OrderId> {
-        self.comit
-            .make_order(order, refund_identity, redeem_identity)
+        let transient = self
+            .storage
+            .derive_transient_identity(swap_id, Role::Alice, Side::Beta);
+
+        self.bitcoin_addresses.insert(transient, refund_identity);
+
+        let data = LocalData {
+            secret_hash: None,
+            ethereum_identity: Some(redeem_identity),
+            bitcoin_identity: Some(transient),
+            lightning_identity: None,
+        };
+        self.local_data.insert(swap_id, data);
+
+        // TODO: Consider adding peer_id to swarm?
```

Comment:
The swarm already knows its peer id.

Reply:
The code comment is stale, removed, thanks.